server/src/main/java/org/opensearch/index/shard/StoreRecovery.java

```diff
@@ -407,6 +407,7 @@ void recoverFromSnapshotAndRemoteStore(
                     threadPool
                 );
```

Comment:
Along with the doc changes of this feature, we need to update doc for `source_remote_store_repository` which will be now used to override remote segment metadata directory: https://opensearch.org/docs/latest/api-reference/snapshots/restore-snapshot/